src/engraving/rendering/dev/passresetlayoutdata.cpp

```diff
@@ -37,8 +39,19 @@ static void resetLayoutData(EngravingItem* item)
     }
 }
 
-void PassResetLayoutData::doRun(Score* score, LayoutContext&)
+void PassResetLayoutData::doRun(Score* score, LayoutContext& ctx)
 {
-    RootItem* rootItem = score->rootItem();
-    resetLayoutData(rootItem);
+    if (ctx.state().isLayoutAll()) {
+        resetLayoutData(score->rootItem());
+    } else {
+        MeasureBase* m = ctx.mutState().nextMeasure();
+        resetLayoutData(m);
+
+        while (m) {
+            m = m->next();
+            if (m && m->tick() < ctx.state().endTick()) {
+                resetLayoutData(m);
+            }
+        }
```

Comment:
I'd do this perhaps a bit differently:
```suggestion
        MeasureBase* m = ctx.mutState().nextMeasure();

        while (m && m->tick() < ctx.state().endTick()) {
            resetLayoutData(m);
            m = m->next();
        }
```